pkg/raft/raft_test.go

```diff
@@ -2178,82 +2178,100 @@ func TestLeaderIncreaseNext(t *testing.T) {
+	r := newTestRaft(1, 10, 1, newTestMemoryStorage(withPeers(1, 2)),
+		withFortificationDisabled())
 
-			r := newTestRaft(1, 10, 1, newTestMemoryStorage(withPeers(1, 2)),
-				testOptions)
+	r.becomeCandidate()
+	r.becomeLeader()
 
-			r.becomeCandidate()
-			r.becomeLeader()
+	r.readMessages()
+	r.trk.Progress(2).BecomeProbe()
 
-			r.readMessages()
-			r.trk.Progress(2).BecomeProbe()
-
-			// each round is a heartbeat
-			for i := 0; i < 3; i++ {
-				if i == 0 {
-					// we expect that raft will only send out one msgAPP on the first
-					// loop. After that, the follower is paused until a heartbeat response
-					// is received.
-					mustAppendEntry(r, pb.Entry{Data: []byte("somedata")})
-					r.maybeSendAppend(2)
-					msg := r.readMessages()
-					assert.Len(t, msg, 1)
-					assert.Zero(t, msg[0].Index)
-				}
+	// each round is a heartbeat
+	for i := 0; i < 3; i++ {
+		if i == 0 {
+			// we expect that raft will only send out one msgAPP on the first
+			// loop. After that, the follower is paused until a heartbeat response
+			// is received.
+			mustAppendEntry(r, pb.Entry{Data: []byte("somedata")})
+			r.maybeSendAppend(2)
+			msg := r.readMessages()
+			assert.Len(t, msg, 1)
+			assert.Zero(t, msg[0].Index)
+		}
 
-				assert.True(t, r.trk.Progress(2).MsgAppProbesPaused)
-				for j := 0; j < 10; j++ {
-					mustAppendEntry(r, pb.Entry{Data: []byte("somedata")})
-					r.maybeSendAppend(2)
-					assert.Empty(t, r.readMessages())
-				}
+		assert.True(t, r.trk.Progress(2).MsgAppProbesPaused)
+		for j := 0; j < 10; j++ {
+			mustAppendEntry(r, pb.Entry{Data: []byte("somedata")})
+			r.maybeSendAppend(2)
+			assert.Empty(t, r.readMessages())
+		}
 
-				// do a heartbeat
-				for j := 0; j < r.heartbeatTimeout; j++ {
-					r.tick()
-				}
-				assert.True(t, r.trk.Progress(2).MsgAppProbesPaused)
-
-				// consume the heartbeat, and the MsgApp if storeliveness is enabled
-				msg := r.readMessages()
-				if storeLivenessEnabled {
-					assert.Len(t, msg, 3)
-					assert.Equal(t, pb.MsgHeartbeat, msg[0].Type)
-					assert.Equal(t, pb.MsgFortifyLeader, msg[1].Type)
-					assert.Equal(t, pb.MsgApp, msg[2].Type)
-				} else {
-					assert.Len(t, msg, 1)
-					assert.Equal(t, pb.MsgHeartbeat, msg[0].Type)
-				}
-			}
+		// do a heartbeat
+		for j := 0; j < r.heartbeatTimeout; j++ {
+			r.tick()
+		}
+		assert.True(t, r.trk.Progress(2).MsgAppProbesPaused)
 
-			// The next heartbeat timeout will allow another message to be sent.
-			for ticks := r.heartbeatTimeout; ticks > 0; ticks-- {
-				r.tick()
-			}
+		// No MsgApp gets sent since we haven't received a MsgHeartbeatResp.
+		msg := r.readMessages()
+		assert.Len(t, msg, 1)
+		assert.Equal(t, pb.MsgHeartbeat, msg[0].Type)
+	}
+
+	// a MsgHeartbeatResp will allow another message to be sent
+	r.Step(pb.Message{From: 2, To: 1, Type: pb.MsgHeartbeatResp})
+	msg := r.readMessages()
+	assert.Len(t, msg, 1)
+	assert.Equal(t, msg[0].Type, pb.MsgApp)
+	assert.Zero(t, msg[0].Index)
```

Comment:
The `Index` is zero because this message batch starts after index 0? Might be worth initializing the raft log at some non-zero state, so that this check is more meaningful.